allennlp/data/dataset_readers/huggingface_datasets_reader.py

```diff
@@ -0,0 +1,276 @@
+import typing
+from typing import Iterable, Optional
+
+from allennlp.data import DatasetReader, Token, Field, Tokenizer
+from allennlp.data.fields import TextField, LabelField, ListField
+from allennlp.data.instance import Instance
+from datasets import load_dataset, DatasetDict, Split, list_datasets
+from datasets.features import ClassLabel, Sequence, Translation, TranslationVariableLanguages
+from datasets.features import Value
+
+
+@DatasetReader.register("huggingface-datasets")
+class HuggingfaceDatasetReader(DatasetReader):
+    """
+    Reads instances from the given huggingface supported dataset
+
+    This reader implementation wraps the huggingface datasets package
+
+    Following dataset and configurations have been verified and work with this reader
+
+            Dataset                       Dataset Configuration
+            `xnli`                        `ar`
+            `xnli`                        `en`
+            `xnli`                        `de`
+            `xnli`                        `all_languages`
+            `glue`                        `cola`
+            `glue`                        `mrpc`
+            `glue`                        `sst2`
+            `glue`                        `qqp`
+            `glue`                        `mnli`
+            `glue`                        `mnli_matched`
+            `universal_dependencies`      `en_lines`
+            `universal_dependencies`      `ko_kaist`
+            `universal_dependencies`      `af_afribooms`
+            `swahili`                     `NA`
+            `conll2003`                   `NA`
+            `dbpedia_14`                  `NA`
+            `trec`                        `NA`
+            `emotion`                     `NA`
+             Note: universal_dependencies will require you to install `conllu` package separately
```

Comment:
Do you think we'll need this list? I'd sort of hope that when it's done, most datasets will work, especially new ones that don't exist yet today. Then maybe we'd have a list for the exceptions, if at all?

Reply:
Right now that is not true, hence I have kept the list. 
When we have larger coverage(most features and nesting support) we would go to a known unsupported datasets.

Reply:
We would need to be able to frequently (i.e before each release) test against all HF datasets to do that. 
I can add a script to do that in a follow up .

Reply:
I don't think we have to test all datasets all the time? Just a few that have some properties that we know make a difference in the code.

Reply:
We would need to go through datasets and pick them for those qualities and from what I have seen it will be quite some amount of work. 877 unique ones right now. 

Yes, not adding this list would mean they try it out and raise issue that it does not work. That is good, since it will allow us to improve it. 
Removing this.